README.md
#JV jQuery Mobile Menu v2
Mobile menu with hamburger button based on jQuery and GSAP JS.<br/>
[View demo](http://archive.juliusvaart.com/jvmobilemenu/)

##Add jQuery, GSAP TweenMax and the plugin just before the closing /body tag
```HTML
<script src="http://code.jquery.com/jquery-1.11.2.min.js"></script>
<script src="http://cdnjs.cloudflare.com/ajax/libs/gsap/latest/TweenMax.min.js"></script>
<script src="jvmobilemenu/jv-jquery-mobile-menu-min.js"></script>
```

##Add the CSS in the head
```HTML
<link rel="stylesheet" type="text/css" href="jvmobilemenu/jv-jquery-mobile-menu.css">
```

##Call the plugin from a JS file and edit the settings
```js
$.jvmobilemenu({
  mainContent: $('.page'),
  theMenu: $('.mobile-nav'),
  slideSpeed: 0.3,
  menuWidth: 240,
  position: 'right',
  menuPadding: '20px 20px 60px'
});
```

##The settings
mainContent	(The Content Wrapper jQuery selector)<br/>
default: ```$('.page')```

theMenu	(Your mobile menu)<br/>
default: ```$('.mobile-nav')```

slideSpeed (Animation speed)<br/>
default: ```0.3``` (seconds)

menuWidth	(Desired menu width)<br/>
default: ```240```

position (Left or right side for hamburger and menu)<br/>
default: ```'right'```

menuPadding	(Padding for the menu content)<br/>
default: ```'20px 20px 60px'```


CHANGELOG.md
#Changelog

##v2.3
Changed hamburger appearance and a little better performance

##v2.2
Better handling of hamburger cross on left side

##v2.1
Added mobile meta to the index.html
Added demo page

##v2.0
Insert hamburger menu html with jQuery
Left or right setting
Easier to use
Plug and play
Something like that...

jvmobilemenu/jv-jquery-mobile-menu-min.js
!function($){$.fn.jvmobilemenu=function(e){function t(){s.removeClass("open"),TweenMax.to(r,settings.slideSpeed/2,{rotation:0,ease:Power3.easeOut}),TweenMax.to(settings.mainContent,settings.slideSpeed,{marginLeft:0}),"left"===settings.position&&TweenMax.to(s,settings.slideSpeed,{marginLeft:i}),setTimeout(function(){settings.theMenu.css({display:"none"})},200),settings.theMenu.css({"overflow-y":"hidden","-webkit-overflow-scrolling":"inherit","overflow-scrolling":"inherit"}),$(document).off("touchmove"),$("body").css({overflow:"inherit"})}function n(){s.addClass("open"),TweenMax.to(a,settings.slideSpeed/2,{rotation:45,ease:Power3.easeOut}),TweenMax.to(d,settings.slideSpeed/2,{rotation:-45,ease:Power3.easeOut}),TweenMax.to(settings.mainContent,settings.slideSpeed,{marginLeft:theMarginLeft}),"left"===settings.position&&TweenMax.to(s,settings.slideSpeed,{marginLeft:theMarginLeft-o}),settings.theMenu.css({display:"block"});var e=".mobile-menu",t=$("body");t.css({overflow:"hidden"}),$(document).on("touchmove",function(e){e.preventDefault()}),t.on("touchstart",e,function(e){0===e.currentTarget.scrollTop?e.currentTarget.scrollTop=1:e.currentTarget.scrollHeight===e.currentTarget.scrollTop+e.currentTarget.offsetHeight&&(e.currentTarget.scrollTop-=1)}),t.on("touchmove",e,function(e){e.stopPropagation()}),settings.theMenu.css({"overflow-y":"scroll","overflow-scrolling":"touch","-webkit-overflow-scrolling":"touch"})}settings=$.extend({mainContent:$(".page"),theMenu:$(".mobile-nav"),slideSpeed:.3,menuWidth:240,position:"right",push:!0,menuPadding:"20px 20px 60px"},e),$(this).prepend('<div class="hamburger"><div class="hamburger-inner"><div class="bar bar1 hide"></div><div class="bar bar2 cross"></div><div class="bar bar3 cross hidden"></div><div class="bar bar4 hide"></div></div></div>'),settings.theMenu.css({width:settings.menuWidth,position:"fixed",top:0,display:"none",height:"100%"}).addClass("mobile-menu").wrapInner('<div class="mobile-menu-inner"></div>'),$(".mobile-menu-inner").css({width:"auto",padding:settings.menuPadding,display:"block"});var s=$(".hamburger"),i=parseInt(s.css("margin-left")),o=s.outerWidth(!0)-i,r=$(".bar2,.bar3"),a=$(".bar2"),d=$(".bar3");"left"===settings.position?(theMarginLeft=settings.menuWidth,settings.theMenu.add(s).css({left:0,right:"auto"})):"right"===settings.position&&(theMarginLeft=-settings.menuWidth,settings.theMenu.add(s).css({left:"auto",right:0})),$(window).resize(function(){t()}),s.on("click",function(e){return $(this).hasClass("open")?t():n(),e.stopPropagation(),!1}),settings.mainContent.on("click",function(){s.hasClass("open")&&t()})},$.jvmobilemenu=function(e){return $("body").jvmobilemenu(e)}}(jQuery);

jvmobilemenu/jv-jquery-mobile-menu.js
/*

  JV jQuery Mobile Menu
  Author: Julius van der Vaart (http://juliusvaart.com)
  Version: 2.3

*/

(function ($) {
  
  $.fn.jvmobilemenu = function (options) {
  
    settings = $.extend({
      // Default settings
      mainContent: $('.page'),
      theMenu: $('.mobile-nav'),
      slideSpeed: 0.3,
      menuWidth: 240,
      position: 'right',
      push: true,
      menuPadding: '20px 20px 60px'
    }, options );
    
    
    // Insert hamburger button
    $(this).prepend('<div class="hamburger"><div class="hamburger-inner"><div class="bar bar1 hide"></div><div class="bar bar2 cross"></div><div class="bar bar3 cross hidden"></div><div class="bar bar4 hide"></div></div></div>');
    
    
    // Menu settings
    settings.theMenu.css({
      width: settings.menuWidth, 
      position: 'fixed',
      top: 0,
      display: 'none',
      height: '100%'
    }).addClass('mobile-menu').wrapInner('<div class="mobile-menu-inner"></div>');
    
    $('.mobile-menu-inner').css({
      width: 'auto', 
      padding: settings.menuPadding, 
      display: 'block'
    });
    
    
    // Hamburger & Mobile Menu vars
    var hamburger = $('.hamburger'),
    hamburgerMarginLeft = parseInt(hamburger.css('margin-left')),
    hamburgerLeftPushPosition = hamburger.outerWidth(true) - hamburgerMarginLeft,
  	crosses = $('.bar2,.bar3'),
  	crossLeft = $('.bar2'),
  	crossRight = $('.bar3');
    
    
    // Mobile menu & hamburger position left or right
    if (settings.position === 'left') {
      theMarginLeft = settings.menuWidth;
      settings.theMenu.add(hamburger)
        .css({
          left: 0, 
          right: 'auto'
        });
    } else if (settings.position === 'right') {
      theMarginLeft = -settings.menuWidth;
      settings.theMenu.add(hamburger)
        .css({
          left: 'auto', 
          right: 0
        });
    }
  
  
  	// menuClose function
  	function menuClose() {
  		
  		// Hamburger
  		hamburger.removeClass('open');
  		
  		//Cross
      TweenMax.to(crosses, settings.slideSpeed / 2, {rotation:0, ease:Power3.easeOut});
  		
  		// Move content back to hide menu
  		TweenMax.to(settings.mainContent, settings.slideSpeed, {marginLeft: 0});
  		
  		if (settings.position === 'left') {
        TweenMax.to(hamburger, settings.slideSpeed, {marginLeft: hamburgerMarginLeft});
      }
  		
  		// Hide content (safari bounce fix)
  		setTimeout( function(){ 
        settings.theMenu.css({display: 'none'}); 
      }, 200);
  		
  		// Disable scrolling plus fix menu-scrolling
  		// From http://stackoverflow.com/a/14244680
  		settings.theMenu.css({
    		'overflow-y': 'hidden',
    		'-webkit-overflow-scrolling': 'inherit',
    		'overflow-scrolling': 'inherit'
      });
  		$(document).off('touchmove');
  		$('body').css({overflow: 'inherit'});
  	}
    
  
  	// menuOpen function
  	function menuOpen() {

    	// Hamburger
  		hamburger.addClass('open');
  		
  		// Cross
  		TweenMax.to(crossLeft, settings.slideSpeed / 2, {rotation:45, ease:Power3.easeOut});
  		TweenMax.to(crossRight, settings.slideSpeed / 2, {rotation:-45, ease:Power3.easeOut});
      
  		// Move content to show menu
      TweenMax.to(settings.mainContent, settings.slideSpeed, {marginLeft: theMarginLeft});
      
      if (settings.position === 'left') {
        TweenMax.to(hamburger, settings.slideSpeed, {marginLeft: theMarginLeft - hamburgerLeftPushPosition});
      }
  		
  		// Show content (safari bounce fix)
  		settings.theMenu.css({display: 'block'});
  		
  		// Disable scrolling on page except header
  		// From http://stackoverflow.com/a/14244680
  		var setScrollable = '.mobile-menu',
  		bodySelect = $('body');
  
  		bodySelect.css({overflow: 'hidden'});
  
  		$(document).on('touchmove',function(e){
  		  e.preventDefault();
  		});
  		bodySelect.on('touchstart', setScrollable, function(e) {
  		if (e.currentTarget.scrollTop === 0) {
  			e.currentTarget.scrollTop = 1;
  		} else if (e.currentTarget.scrollHeight === e.currentTarget.scrollTop + e.currentTarget.offsetHeight) {
  			e.currentTarget.scrollTop -= 1;
  		}
  		});
  		bodySelect.on('touchmove', setScrollable, function(e) {
  		  e.stopPropagation();
  		});
  		settings.theMenu.css({
    		'overflow-y': 'scroll',
    		'overflow-scrolling': 'touch',
    		'-webkit-overflow-scrolling': 'touch'
      });
  	}
  
  
  	// Close menu on window-resize
  	$(window).resize(function() {
  		menuClose();
  	});
  	
  
  	// Hamburger click
  	hamburger.on('click', function(e) {
  		if ($(this).hasClass('open')) {
  			menuClose();
  		} else {
  			menuOpen();
  		}
  		e.stopPropagation();
  		return false;
  	});
  
  
  	// Close main-menu on click outside menu
  	settings.mainContent.on('click', function() {
    	if (hamburger.hasClass('open')) {
  			menuClose();
  		}
  	});
  
  };
  
  // If no element is supplied, we'll attach to body
  // Borrowed from https://github.com/srobbin/jquery-backstretch
  $.jvmobilemenu = function (options) {
    // Return the instance
    return $('body').jvmobilemenu(options);
  };

})(jQuery);